Dockerfile

```diff
@@ -1,15 +1,15 @@
-FROM golang:1.9
+FROM golang:1.9-alpine
 
 COPY cmd/freegeoip/public /var/www
 
 ADD . /go/src/github.com/apilayer/freegeoip
-RUN \
-	cd /go/src/github.com/apilayer/freegeoip/cmd/freegeoip && \
-	go get -d && go install && \
-	apt-get update && apt-get install -y libcap2-bin && \
-	setcap cap_net_bind_service=+ep /go/bin/freegeoip && \
-	apt-get clean && rm -rf /var/lib/apt/lists/* && \
-	useradd -ms /bin/bash freegeoip
+WORKDIR /go/src/github.com/apilayer/freegeoip/cmd/freegeoip
+RUN apk update
+RUN apk add git libcap shadow
+RUN go get -d
```

Comment:
Why not group all of these `RUN`s into one layer like the existing image?

Reply:
Splitting it up (from my limited knowlege) allows if you change the lower ones the upper ones will be cached, meaning you won't have to run the higher ones nearly as often. Perhaps I missunderstood that though.

Reply:
In my most recent PR I've organized it so that it _should_ cache installing the depenecies (changes infrequently). I found what helped the most was moving the COPY down. However It seems from this discussion that there's tradofs of caching for file size differences: https://stackoverflow.com/questions/39223249/multiple-run-vs-single-chained-run-in-dockerfile-what-is-better